### PR TITLE
[Merged by Bors] - add Visibility for lights

### PR DESCRIPTION
**Pull request by robtfm**
# Objective

Add Visibility for lights

## Solution

- add Visibility to PointLightBundle and DirectionLightBundle
- filter lights used by Visibility.is_visible

note: includes changes from #3916 due to overlap, will be cleaner after that is merged

### CODE REVIEW

File: crates/bevy_pbr/src/light.rs [hunk 1]
```diff
@@ -480,14 +481,64 @@ fn cluster_to_index(cluster_dimensions: UVec3, cluster: UVec3) -> usize {
     ((cluster.y * cluster_dimensions.x + cluster.x) * cluster_dimensions.z + cluster.z) as usize
 }
 
+// Sort point lights with shadows enabled first, then by a stable key so that the index
+// can be used to render at most `MAX_POINT_LIGHT_SHADOW_MAPS` point light shadows, and
```
Comment (alice-i-cecile) [inline crates/bevy_pbr/src/light.rs:485]:
```suggestion
// can be used to render at most `MAX_POINT_LIGHT_SHADOW_MAPS` point light shadows and
```

File: crates/bevy_pbr/src/light.rs [hunk 2]
```diff
@@ -506,7 +557,7 @@ pub fn assign_lights_to_clusters(
             vec![VisiblePointLights::from_light_count(light_count); cluster_count];
         let mut visible_lights_set = HashSet::with_capacity(light_count);
 
-        for (light_entity, light_transform, light) in lights.iter() {
+        for &(light_entity, light_transform, light, _) in lights.iter() {
```
Comment (alice-i-cecile) [inline crates/bevy_pbr/src/light.rs:560]: This suggests that we should be using a `With` filter above.

Comment (alice-i-cecile) [inline crates/bevy_pbr/src/light.rs:560]:
```suggestion
        for &(light_entity, light_transform, light, _visible) in lights.iter() {
```

**Review by superdump — APPROVED** [review]
I think this LGTM when #3916 is merged. I'll re-check it once that has been.

File: crates/bevy_pbr/src/light.rs [hunk 3]
```diff
@@ -480,7 +480,7 @@ fn ndc_position_to_cluster(
 }
 
 // Sort point lights with shadows enabled first, then by a stable key so that the index
-// can be used to render at most `MAX_POINT_LIGHT_SHADOW_MAPS` point light shadows, and
+// can be used to render at most `MAX_POINT_LIGHT_SHADOW_MAPS` point light shadows and
```
Comment (superdump) [inline crates/bevy_pbr/src/light.rs:483]: I tend to keep these Oxford commas these days but I won’t block on its removal.

**Review by superdump — APPROVED** [review]

**Review by IceSentry — COMMENTED** [review]
Taking into consideration what @superdump said. I think just adding a short doc comment on the property is more than enough.

**Review by IceSentry — APPROVED** [review]

**Review by mockersf — APPROVED** [review]

**Pull request closed without merging**